garden_ai/model.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+from datetime import datetime
+from enum import Enum, auto
+from typing import List
+from uuid import uuid4, UUID
+import pydantic
+from pydantic import BaseModel, Field, validator
+
+
+class TempField(Enum):
+    REQUIRED = auto()  # just those needed to get a doi
+    WARN = auto()
+    # setting this should trigger warning before publishing the model
+    # we can also inspect the object to see which fields have been set manually
+    # might be more flexible in the future
+
+
+class Garden(BaseModel):
+    """Object representation of a Garden™.
+
+    Required Attributes
+    ----------
+    creators: List[str]
+        The main researchers involved in producing the Garden. At least one
+        creator is required in order to register a DOI. Personal name format
+        should be: "Family, Given".
+
+    title: str
+        An official name or title for the Garden. This attribute must be set
+        in order to register a DOI.
+
+
+    Recommended Attributes
+    ----------
+
+    description: str
+        A brief summary of the Garden and/or its purpose, to aid discovery by
+        other Gardeners.
+
+
+    Optional Attributes
+    ----------
+    doi: str
+        A garden's doi can be manually set by the user, or generated automatically via the
+        DataCite rest api using the required fields.
+
+    language: str
```

Comment:
Nice, but add clarity on what this means -- I assume it's just a metadata field explaining what language the abstract, etc are written in?

Otherwise, it could be read as a field for setting the language (ie we do translation)